internal/datanode/compaction/mix_compactor.go

```diff
@@ -184,6 +184,7 @@ func (t *mixCompactionTask) mergeSplit(
 					break
```

Comment:
Better to add a defer func after NewBinlogDesirializedReader at Line:175 for better maintainability, instead of adding Close() everywhere and worry about missing in somewhere else

```golang
defer iter.Close()
```

Reply:
It is not recommended to use `defer` in for-loop
Maybe we could use the trick that use a closure like
```
for ... {
    func() {
        iter, err := NewIterator(...)
        if err != nil {}
        defer iter.Close()
    }()
}
```

Reply:
> Better to add a defer func after NewBinlogDesirializedReader at Line:175 for better maintainability, instead of adding Close() everywhere and worry about missing in somewhere else
> 
> ```go
> defer iter.Close()
> ```

The best way i think is to define a new func with defer close() contains from line 174 to line 210, instead of user defer in for-loop

```go
for _, paths := range binlogPaths {
      dealBinlogPath()
}

dealBinlogPath() {
	iter, err := storage.NewBinlogDeserializeReader(blobs, pkField.GetFieldID())
	if err != nil {
		log.Warn("compact wrong, failed to new insert binlogs reader", zap.Error(err))
		return nil, err
	}
	defer iter.Close()
	// other logic
}
```

Reply:
@fengjun2016 that looks good to me. It's OK to make `dealBinlogPath` a lambda.